enumerable.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    temp = self
+    i = 0
+    while i < temp.length
+      yield(temp[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    temp = self
+    i = 0
+    while i < temp.length
+      yield(temp[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    temp = []
+    my_each { |e| temp << e if yield(e) }
+    temp
+  end
+
+  def my_all?
+    test = true
+    my_each { |d| break unless test == yield(d) }
+    test
+  end
```

Comment:
In `my_all?`
- enumerables my_all? returns true if the block never returns false or nil
rspec ./spec/spec_enumerables_spec.rb:69 # enumerables my_all? when no block or argument is given returns true when none of the collection members are false or nil
- enumerables my_all? when a class is passed as an argument returns true if all of the collection is a member of such class
- enumerables my_all? when a Regex is passed as an argument returns true if all of the collection matches the Regex
- enumerables my_all? when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern